### PR TITLE
Expoential as a default reconnection policy

### PR DESCRIPTION
**Pull request by seba-aln**

### CODE REVIEW

File: pubnub/managers.py [hunk 1]
```diff
@@ -89,6 +93,26 @@ def _stop_heartbeat_timer(self):
             self._timer = None
 
 
+class LinearDelay:
```
Comment (jguz-pubnub) [inline pubnub/managers.py:96]: Is it possible to provide custom delay other than `2`?

File: pubnub/pubnub.py [hunk 2]
```diff
@@ -101,6 +101,13 @@ def __init__(self, pubnub):
     def _register_heartbeat_timer(self):
         self.stop_heartbeat_timer()
 
+        if self._retry_limit_reached():
```
Comment (jguz-pubnub) [inline pubnub/pubnub.py:104]: I don't know how it works in Python, but should announcing `disconnected` status being part of EE?

File: tests/acceptance/subscribe/steps/when_steps.py [hunk 3]
```diff
@@ -4,12 +4,14 @@
...
-def step_impl(context: PNContext):
+@async_run_until_complete
+async def step_impl(context: PNContext):
     context.pubnub.subscribe().channels('foo').execute()
 
 
 @when('I subscribe with timetoken {timetoken}')
-def step_impl(context: PNContext, timetoken: str): # noqa F811
+@async_run_until_complete
+async def step_impl(context: PNContext, timetoken: str): # noqa F811
```
Comment (jguz-pubnub) [inline tests/acceptance/subscribe/steps/when_steps.py:14]: I'm curious, what it means? `# noqa F811`

File: pubnub/managers.py (hunk 1 above)
Comment (seba-aln) [reply pubnub/managers.py:96]: No, not really. Should we add this option?

File: pubnub/pubnub.py (hunk 2 above)
Comment (seba-aln) [reply pubnub/pubnub.py:104]: No. This one is for non-event engine subscribe (yeah, python has more than one subscribe)

File: tests/acceptance/subscribe/steps/when_steps.py (hunk 3 above)
Comment (seba-aln) [reply tests/acceptance/subscribe/steps/when_steps.py:14]: it disables validation errors upon redeclaration of method. in this file there are more methods called `step_impl`, but they are the content of a `@when` decorator. so TL:DR - 'tis just for linter

File: pubnub/managers.py (hunk 1 above)
Comment (seba-aln) [reply pubnub/managers.py:101]: added in latest commit. thanks for pointing that out ;)

**Review by jguz-pubnub — APPROVED** [review]

**Pull request merged**